data-raw/testdata.R

```diff
@@ -0,0 +1,271 @@
+# Script to generate synthetic data for tests
+
+# Load required libraries
+library(tidyverse)
+library(data.table)
+library(here)
+library(lubridate)
+
+# Set seed for reproducibility
+set.seed(123)
+
+# Set number of rows to create for fake data
+number_rows <- 1000
+
+
+# Functions ---------------------------------------------------------------
+
+generate_fake_atc <- function(number_rows) {
+  codeCollection::ATCKoodit |>
+    tibble::as_tibble() |>
+    dplyr::filter(stringr::str_length(Koodi) == 7) |>
+    dplyr::pull(Koodi) |>
+    sample(number_rows, replace = TRUE)
+}
+
+generate_fake_indication <- function(number_rows) {
+  sample(1:9e8, number_rows, replace = TRUE) |>
+    stringr::str_trunc(width = 7, ellipsis = "") |>
+    stringr::str_pad(width = 7, pad = "0")
+}
+
+assign_drugname_from_atc <- function(data) {
+  codeCollection::ATCKoodit |>
+    tibble::as_tibble() |>
+    dplyr::select(ATC = Koodi, drugname = en) |>
+    dplyr::right_join(data, by = "ATC", relationship = "many-to-many")
+}
+
+# Medication data ---------------------------------------------------------
+
+# Pseudo-lmdb:
+
+## Non-diabetes data:
+
+# Create a tibble with 1000 rows from 200 individuals
+med_df <- tibble(
+  # ID variable
+  pnr = sample(sprintf("%03d", 1:200), number_rows, replace = TRUE),
+  # Date of purchase
+  eksd = as_date(sample(
+    seq(as_date("1995-01-01"), as_date("2014-12-31"), by = "day"), 1000,
+    replace = TRUE
+  )),
+  # Number of packages
+  apk = sample(1:3, 1000, replace = TRUE),
+  # Indication for treatment
+  indo = generate_fake_indication(number_rows),
+  # ATC code
+  ATC = generate_fake_atc(number_rows),
+  # Volume
+  volume = sample(20:100, 1000, replace = TRUE)
```

Comment:
What is this for?

Reply:
volume is a measure of how many doses are contained in the package purchased. It is used to classify type 1 from type 2 diabetes (evaluating insulin purchases vs. non-insulin glucose-lowering drugs purchased).